server/pulp/server/managers/content/orphan.py

```diff
@@ -236,10 +237,16 @@ def delete_orphan_content_units_by_type(type_id):
 
         :param type_id: id of the content type
         :type type_id: basestring
+        :param content_unit_ids: list of content unit ids to delete; None means delete them all
+        :type content_unit_ids: iterable or None
         """
         # get the model matching the type
         content_model = plugin_api.get_unit_model_by_id(type_id)
-        content_units = content_model.objects().only('id', '_storage_path')
+        if content_unit_ids:
+            query_set = content_model.objects(id__in=content_unit_ids)
```

Comment:
Although it is *very* unlikely that anyone would do that, it is technically possible.